### PR TITLE
Code clean up after prod backfill

### LINKED ISSUE #9594: Data backfills | Code cleanup
As we finish our data backfills, we can update our code to expect the data to be in the right state.

1) In schedule_hearing_task.rb, we can update find_or_create_if_eligible to just find the existing ScheduleHearingTask
1) In tasks_controller.rb, we can remove find_or_create_if_eligible from the for_appeal method
1) In hold_hearing_task.rb, we can remove the type check on the task's parent in create_hold_hearing_task!
1) In appeal_repository.rb, we can update create_schedule_hearing_tasks to always create a new HearingTask and a new ScheduleHearingTask
1) In ScheduleHearingTask and DispositionTask (formerly HoldHearingTask), we can add validation such that the parent task always has to be a HearingTask

### PR DESCRIPTION
**Pull request by oscarAtNava**
Resolves #9594 

### Description
Remove some code no longer needed now that prod data is cleaned up.

### Acceptance Criteria
- [ ] Code compiles correctly

### Testing Plan
1. TBD

### CODE REVIEW

File: app/services/appeal_repository.rb [hunk 1]
```diff
@@ -345,11 +345,9 @@ def self.create_schedule_hearing_tasks
...
         parent: RootTask.find_or_create_by!(appeal: appeal, assigned_to: Bva.singleton)
       ) { |task| task.assigned_to = Bva.singleton }
-      if ScheduleHearingTask.where(appeal: appeal).where.not(status: Constants.TASK_STATUSES.completed).empty?
-        ScheduleHearingTask.create!(appeal: appeal) do |task|
-          task.assigned_to = HearingsManagement.singleton
-          task.parent = parent
-        end
+      ScheduleHearingTask.create!(appeal: appeal) do |task|
+        task.assigned_to = HearingsManagement.singleton
+        task.parent = parent
```
Comment (sharonwarner) [inline app/services/appeal_repository.rb]: This should be updated to always create a new HearingTask as well.

Comment (oscarAtNava) [reply app/services/appeal_repository.rb]: Do you mean to replace HearingTask.find_or_create_by!() with just HearingTask.create!()?

Comment (sharonwarner) [reply app/services/appeal_repository.rb]: Yes, that's correct!

**Comment by tomas-nava** [comment]
This PR also resolves #9538

**Comment by oscarAtNava** [comment]
@sharonwarner Almost all Disposition and ScheduleHearingTasks have a HearingTask parent. I ran the following SQL Queries to confirm and found one issue:

### Total ScheduleHearingTask and DispositionTask
select count(*) 
from tasks
where type IN ('DispositionTask', 'ScheduleHearingTask')

Result ==> 28,190

### DispositionTask and ScheduleHearingTask with HearingTask parent
select count(*) 
from tasks p, tasks c
where c.parent_id = p.id
and c.type IN ('DispositionTask', 'ScheduleHearingTask')
and p.type = 'HearingTask'

Results ==> 28,189

Appeal with issue appeal_id = 864. It is an AMA appeal hence was not touched by my script. We coul fix it manually next week.

**Comment by sharonwarner** [comment]
Yes, it's fine to ignore that AMA appeal. It's already been distributed to an attorney, so we shouldn't need to do anything else.

**Pull request merged**